client/cli/src/lib.rs

```diff
@@ -112,3 +112,37 @@ impl sc_cli::CliConfiguration for PurgeChainCmd {
 		Some(&self.base.database_params)
 	}
 }
+
+/// The `run` command used to run a node.
+#[derive(Debug, StructOpt)]
+pub struct RunCmd {
+	/// Our run command inherents from sc_cli's
+	#[structopt(flatten)]
+	pub base: sc_cli::RunCmd,
+
+	/// Id of the parachain this collator collates for.
+	#[structopt(long)]
+	pub parachain_id: Option<u32>,
+
+	/// Run node as collator.
+	///
+	/// Note that this is the same as running with `--validator`.
+	#[structopt(long, conflicts_with = "validator")]
+	pub collator: bool,
+}
+
+impl sc_cli::CliConfiguration for RunCmd {
+	fn shared_params(&self) -> &sc_cli::SharedParams {
+		&self.base.shared_params
+	}
+
+	fn role(&self, is_dev: bool) -> sc_cli::Result<sc_cli::Role> {
+		let is_authority = self.base.validator || self.collator || is_dev;
+
+		Ok(if is_authority {
+			sc_service::Role::Authority
+		} else {
+			sc_service::Role::Full
+		})
+	}
+}
```

Comment:
I'm afraid this won't be enough. You need to call every method that [the original `RunCmd` implements](https://github.com/paritytech/substrate/blob/master/client/cli/src/commands/run_cmd.rs#L269-L445). Just like you did for `shared_params`.

You can get inspiration from this code we do for the relay chain: https://github.com/paritytech/cumulus/blob/master/rococo-parachains/src/command.rs#L349-L457